src/main/kotlin/com/terraformation/backend/documentproducer/db/variable/AllVariableCsvVariable.kt

```diff
@@ -0,0 +1,110 @@
+package com.terraformation.backend.documentproducer.db.variable
+
+import com.terraformation.backend.db.accelerator.DeliverableId
+import com.terraformation.backend.db.docprod.DependencyCondition
+import com.terraformation.backend.db.docprod.VariableId
+import com.terraformation.backend.db.docprod.VariableTableStyle
+import com.terraformation.backend.db.docprod.VariableType
+import com.terraformation.backend.db.docprod.tables.pojos.VariableSelectOptionsRow
+import com.terraformation.backend.db.docprod.tables.pojos.VariablesRow
+import java.math.BigDecimal
+
+private const val LOCALIZED_ERROR_KEY_UNKNOWN_DATA_TYPE = "variablesCsvDataTypeUnknown"
+
+// In the header order in the CSV
+data class AllVariableCsvVariable(
+    /** Column 1/A - Name */
+    val name: String,
+    /** Column 2/B - Identifier that stays stable across settings changes. */
+    val stableId: String,
+    /** Column 3/C - Description (optional) */
+    val description: String?,
+    /** Column 4/D - Data Type */
+    val dataType: AllVariableCsvVariableType,
+    /** Column 5/E - List? */
+    val isList: Boolean,
+    /** Column 6/F - Parent (for non-top-level sections and table columns) */
+    val parent: String?,
+    /**
+     * Column 7/G - Options (for Select; put each value on a separate line, optionally prefixed with
+     * a hyphen; if the value should result in different text in the rendered document, put the text
+     * in double square brackets after the option)
+     */
+    val options: List<VariableSelectOptionsRow>,
+    /** Column 8/H - Minimum value, if any (for Number) */
+    val minValue: BigDecimal?,
+    /** Column 9/I - Maximum value, if any (for Number) */
+    val maxValue: BigDecimal?,
+    /** Column 10/J - Decimal places (for Number) */
+    val decimalPlaces: Int?,
+    /** Column 11/K - Table style (for Table) */
+    val tableStyle: VariableTableStyle?,
+    /** Column 12/L - Header? (for table columns) */
+    val isHeader: Boolean,
+    /** Column 13/M - Notes */
+    val notes: String?,
+    /** Column 14/N - Deliverable ID */
+    val deliverableId: DeliverableId?,
+    /** Column 15/O - Deliverable Question */
+    val deliverableQuestion: String?,
+    /** Column 16/P - Dependency - Variable ID */
+    val dependencyVariableId: VariableId?,
```

Comment:
Wouldn't this need to be a stable ID, not a variable ID? If you're adding a new variable and another new variable that depends on it, there won't be a variable ID for the first one yet.

Reply:
Good point, will fix it up